ibl_to_nwb/AlyxToNWB/io_tools.py

```diff
@@ -0,0 +1,208 @@
+from lazy_ops import DatasetView
+import numpy as np
+import pandas as pd
+from datetime import datetime
+from tzlocal import get_localzone
+
+
+def iter_datasetview(datasetview_obj):
+    '''
+    Generator to return a row of the array each time it is called.
+    This will be wrapped with a DataChunkIterator class.
+
+    Parameters
+    ----------
+    datasetview_obj: DatasetView
+        2-D array to iteratively write to nwb.
+    '''
+
+    for i in range(datasetview_obj.shape[0]//700):
+        curr_data = np.squeeze(datasetview_obj[i:i + 1])
+        yield curr_data
+    return
+
+def get_default_column_ids(default_namelist, namelist):
+    out_idx = []
+    for j, i in enumerate(namelist):
+        if i in default_namelist:
+            out_idx.extend([j])
+    return out_idx
+
+
+class OneData:
+
+    def __init__(self, one_object, eid, no_probes, nwb_metadata, save_raw=False, save_camera_raw=False):
+        self.one_object = one_object
+        self.eid = eid
+        self.no_probes = no_probes
+        self.save_raw = save_raw
+        self.save_camera_raw = save_camera_raw
+        self.loaded_datasets = dict()
+        self.data_attrs_dump = dict()
+        self.nwb_metadata = nwb_metadata
+
+    def download_dataset(self, dataset_to_load, dataset_key):
+        if not type(dataset_to_load) is str:  # prevents errors when loading metafile json
+            return
+        if dataset_to_load.split('.')[0] == 'ephysData' and not self.save_raw:
+            return
+        if dataset_to_load.split('.')[0] == '_iblrig_Camera' and not self.save_camera_raw:
+            return
+        if dataset_to_load not in self.loaded_datasets.keys():
```

Comment:
here and elsewhere, don't iterate over dict.keys(). iterating over the dict does the same thing

Reply:
done https://github.com/catalystneuro/IBL-to-nwb/pull/9/commits/8ac2b06cd462722d15b71c37608ac4fc3b66aff9